packages/core/src/components/TextField/Mask.jsx

```diff
@@ -69,13 +101,30 @@ export class Mask extends React.PureComponent {
    */
   maskedValue(value = '') {
     if (value && typeof value === 'string') {
+      const { mask } = this.props;
       value = value.trim();
 
-      if (this.props.mask === 'currency') {
+      if (mask === 'currency') {
         // Format number with commas. If the number includes a decimal,
         // ensure it includes two decimal points
         value = this.toNumber(value);
         value = this.stringWithFixedDigits(value.toLocaleString('en-US'));
+      } else if (['phone', 'ssn', 'zip'].indexOf(mask) >= 0) {
```

Comment:
We do most of the work for the `currency` branch in another function. If we move this transformation function to someplace external with a persistent `maskRegex` object, we could just reference `Object.keys(maskRegex).includes(mask)` rather than listing the keys again. What if we also moved `stringWithFixedDigits` outside the class since we never reference `this` inside it?

Reply:
It's just that I feel like these string transformations that are `string => string` don't need to be inlined here, and it would decrease the complexity of this function. I like how you did it with the currency block.